org.lflang/src/org/lflang/federated/extensions/FedTargetExtension.java

```diff
@@ -0,0 +1,136 @@
+     * @param errorReporter
+     * @param federationRTIProperties
+     */
+    void createLauncher(
+        List<FederateInstance> federates,
+        FileConfig fileConfig,
+        TargetConfig targetConfig,
+        ErrorReporter errorReporter,
+        LinkedHashMap<String, Object> federationRTIProperties
+    ) throws IOException;
+
+    /**
+     * Generate code for the body of a reaction that handles the
+     * action that is triggered by receiving a message from a remote
+     * federate.
+     * @param action The action.
+     * @param sendingPort The output port providing the data to send.
+     * @param receivingPort The ID of the destination port.
+     * @param connection FIXME
+     * @param type FIXME
+     * @param coordinationType The coordination type
+     * @param errorReporter
+     */
+    String generateNetworkReceiverBody(
+        Action action,
+        VarRef sendingPort,
+        VarRef receivingPort,
+        FedConnectionInstance connection,
+        InferredType type,
+        CoordinationType coordinationType,
+        ErrorReporter errorReporter
+    );
+
+    /**
+     * Generate code for the body of a reaction that handles an output
+     * that is to be sent over the network.
+     * @param sendingPort The output port providing the data to send.
+     * @param receivingPort The variable reference to the destination port.
+     * @param connection
+     * @param type
+     * @param coordinationType
+     * @param errorReporter FIXME
+     */
+    String generateNetworkSenderBody(
+        VarRef sendingPort,
+        VarRef receivingPort,
+        FedConnectionInstance connection,
+        InferredType type,
+        CoordinationType coordinationType,
+        ErrorReporter errorReporter
+    );
+
+    /**
+     * Generate code for the body of a reaction that decides whether the trigger for the given
+     * port is going to be present or absent for the current logical time.
+     * This reaction is put just before the first reaction that is triggered by the network
+     * input port "port" or has it in its sources. If there are only connections to contained
+     * reactors, in the top-level reactor.
+     *
+     * @param receivingPortID The port to generate the control reaction for
+     * @param maxSTP The maximum value of STP is assigned to reactions (if any)
+     *  that have port as their trigger or source
+     * @param coordination FIXME
+     */
+    String generateNetworkInputControlReactionBody(
+        int receivingPortID,
+        TimeValue maxSTP,
+        CoordinationType coordination
+    );
+
+    /**
+     * Generate code for the body of a reaction that sends a port status message for the given
+     * port if it is absent.
+     *
+     * @oaram srcOutputPort FIXME
+     * @param connection FIXME
+     */
+    String generateNetworkOutputControlReactionBody(
+        VarRef srcOutputPort,
+        FedConnectionInstance connection
+    );
+
+    /**
+     * FIXME
```

Comment:
Can you add a brief explanation of this method?

Reply:
Will do (the FIXMEs are temporary placeholders). [Here](https://github.com/lf-lang/lingua-franca/blob/1c7924c2d6833014a4664bd74651d061934fad86/org.lflang/src/org/lflang/generator/GeneratorBase.java#L717) is the old comment for this method (the functionality hasn’t changed so the comment should probably be along the same lines).

Nonetheless, I think it would be better if we only move existing code for now in this PR and not implement new features. Control reactions are an unimplemented feature in the TS target as far as I know.

Reply:
Sorry, I was looking at the wrong thing. Will add a comment shortly.